profiles/seccomp/kernel_linux.go

```diff
@@ -0,0 +1,96 @@
+package seccomp
+
+import (
+	"bytes"
+	"fmt"
+	"sync"
+
+	"golang.org/x/sys/unix"
+)
+
+// kernelVersion holds information about the kernel.
+type kernelVersion struct {
+	kernel uint // Version of the kernel (i.e., the "4" in "4.1.2-generic")
+	major  uint // Major revision of the kernel (i.e., the "1" in "4.1.2-generic")
+}
+
+func (k *kernelVersion) String() string {
+	return fmt.Sprintf("%d.%d", k.kernel, k.major)
+}
+
+var (
+	currentKernelVersion *kernelVersion
+	once                 sync.Once
+)
+
+// getKernelVersion gets the current kernel version.
+func getKernelVersion() (*kernelVersion, error) {
+	var err error
+	once.Do(func() {
+		var uts unix.Utsname
+		err = unix.Uname(&uts)
+		if err != nil {
+			return
+		}
+		// Remove the \x00 from the release for Atoi to parse correctly
+		currentKernelVersion, err = parseRelease(string(uts.Release[:bytes.IndexByte(uts.Release[:], 0)]))
+	})
+	return currentKernelVersion, err
+}
+
+// parseRelease parses a string and creates a kernelVersion based on it.
+func parseRelease(release string) (*kernelVersion, error) {
+	var (
+		version = kernelVersion{}
+		suffix  string
+	)
+
+	// Just make sure we get the "kernel" and "major revision". Sometimes we have
+	// 3.12.25-gentoo, but sometimes we just have 3.12-1-amd64. We add a bogus
+	// "-any-suffix" suffix to account for anything after "kernel.major" being
+	// optional, and may not be present in the input string.
+	parsed, err := fmt.Sscanf(release+"-any-suffix", "%d.%d%s", &version.kernel, &version.major, &suffix)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse kernel version %q: %w", release, err)
+	}
+	if parsed < 2 {
+		return nil, fmt.Errorf("failed to parse kernel version %q", release)
+	}
+	return &version, nil
+}
+
+// parseMinVersion parses a string and creates a kernelVersion based on it. Unlike
+// parseRelease, only "kernel" and "major revision" can be specified (e.g., "3.12").
+func parseMinVersion(minVersion string) (*kernelVersion, error) {
+	var version = kernelVersion{}
+	parsed, err := fmt.Sscanf(minVersion, "%d.%d", &version.kernel, &version.major)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse kernel version %q: %w", minVersion, err)
+	}
+	if parsed < 2 {
+		return nil, fmt.Errorf("failed to parse kernel version %q", minVersion)
+	}
+	return &version, nil
+}
+
+// kernelGreaterEqualThan checks if the host's kernel version is greater than, or
+// equal to the given kernel version v. Only "kernel version" and "major revision"
+// can be specified (e.g., "3.12") and will be taken into account, which means
+// that 3.12.25-gentoo and 3.12-1-amd64 are considered equal (kernel: 3, major: 12).
+func kernelGreaterEqualThan(v string) (bool, error) {
+	minVersion, err := parseMinVersion(v)
+	if err != nil {
+		return false, err
+	}
+	currentKernelVersion, err = getKernelVersion()
```

Comment:
seems you removed the caching for this

Reply:
He has a sync.Once

Reply:
ok, but the error handling doesn't look correct there

Reply:
Yes, the error would only be returned the first time. Looking at it, I think we don't really need the error, and instead could return `false` if we were unable to get the kernel version (in which case `getKernelVersion()` would return `nil`)